Documentation/rfc/v3api.md

```diff
@@ -0,0 +1,459 @@
+## Design
+
+1. Flatten binary key-value space
+    
+2. Keep the event history until compaction
+    - access to old version of keys
+    - user controlled history compaction
+    
+3. Support range query
+    - Pagination support with limit argument
+    
+4. Replace TTL key with Lease
+    - more efficient/ low cost keep alive
+    - a logical group of TTL keys
+    
+5. Replace CAS/CAD with multi-object Tnx
+    - MUCH MORE powerful and flexible
```

Comment:
Yes (with empty compare list). Take a look around Line 191 for the format.

We might limit the total size of a bulk loading.